src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,74 @@
+package lotto.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lotto.domain.Budget;
+import lotto.domain.LottoMachine;
+import lotto.domain.dto.LottoResultDto;
+import lotto.domain.LottoNumber;
+import lotto.domain.dto.TicketDto;
+import lotto.domain.WinningNumbers;
+import lotto.view.LottoView;
+
+public class LottoController {
+    private final LottoMachine lottoMachine;
+    private final LottoView view;
+    private List<LottoNumber> lottoNumberPool;
+    private boolean holdFlag;
+
+    public LottoController() {
+        initNumberPool();
+        this.lottoMachine = new LottoMachine(lottoNumberPool);
+        this.view = new LottoView();
+        this.holdFlag = true;
+    }
+
+    public void start() {
+        List<TicketDto> tickets = null;
+        while (holdFlag) {
+            tickets = createTickets();
+        }
```

Comment:
변수에 값을 할당할 때, 일단 null로 할당한 뒤 추후에 재할당하는 방식이 아니라 변수를 선언함과 동시에 (null이 아닌) 값을 할당하도록하도록 변경해보면 자연스레 해결될 것 같아요. 

추가적으로 `holdFlag`라는 인스턴스 변수가 필요한지 잘 모르겠어요. 🤔(꼭 인스턴스 변수가 아닌 지역변수라도 불필요한 것 같아요)

`createTickets` 메서드가 LottoTickets 생성에 실패한 경우 null을 반환하는 것이 아니라 다시 입력을 요청하도록 구조를 변경해보면 더 깔끔하게 코드를 작성하실 수 있을 것 같아요.

```suggestion
        List<TicketDto> tickets = createTickets();
```

createTickets의 입력이 실패한 경우 `null`을 반환하는 것이 아니라 다시 createTickets를 입력받도록 수정해보면 좋을 것 같아요. 🙂
```java
private List<TicketDto> createTickets() {
        try {
            int budget = view.getBudget();
            return lottoMachine.generateTickets(new Budget(budget));
        } catch (RuntimeException e) {
            view.printError(e);
            return createTickets();
        }
    }
```

아래의 `result`도 지금과 유사하게 변경할 수 있겠네요. 🙂

Reply:
catch 블록에서 재귀적으로 함수를 호출하면 holdFlag 없이도 재입력을 처리할 수 있는군요!
그런데 혹시 아래와 같이 잘못된 입력값을 계속 보내는 경우 메소드 호출 스택이 계속 쌓이게 되면서 최악의 경우 스택 오버플로우가 발생할 수도 있을 것 같은데 어떻게 보완해야 할까요?

<img width="284" alt="image" src="https://github.com/next-step/java-lotto-kakao/assets/100896063/6d1d975b-cd9d-415b-aa68-8f5d792c5bab">
<img width="310" alt="image" src="https://github.com/next-step/java-lotto-kakao/assets/100896063/09e058f6-c860-4c4c-871b-8e85bf45392c">

Reply:
물론 재귀함수는 스택오버 플로우의 문제가 발생할 수 있을거에요. 🙂

먼저 재귀호출은 대략적으로 4800번 조금 넘는 정도로 재귀호출할 경우 `StackOverflowError`가 발생하는 것으로 보여요. 

사실 저는 4800번 정도로 실패하는 케이스는 극히 드문 엣지 케이스라고 생각되는데, 그럼에도 이런 케이스를 방지하겠다면 
재귀함수 대신에 while문을  쓰는 방식으로 처리할 수 있을 것 같네요. 🙂 
동현님이 처음에 쓰셨던 것 처럼 flag를 써볼 수 있을 것 같아요. 단지 flag를 Controller의 인스턴스 변수로 두는 것 보다는 지역변수 정도로 둘 것 같네요. 🙂 사실 그 외에도 더 좋은 방법이 있을지는 모르겠지만 사실 실패 케이스가 어느정도 누적되면 적당선에서 애플리케이션을 종료하도록하는 것도 하나의 방법일 것 같아요.

Reply:
물론 재귀함수는 스택오버 플로우의 문제가 발생할 수 있을거에요. 🙂

먼저 재귀호출은 대략적으로 4800번 조금 넘는 정도로 재귀호출할 경우 `StackOverflowError`가 발생하는 것으로 보여요. 

사실 저는 4800번 정도로 실패하는 케이스는 극히 드문 엣지 케이스라고 생각되는데, 그럼에도 이런 케이스를 방지하겠다면 
재귀함수 대신에 while문을  쓰는 방식으로 처리할 수 있을 것 같네요. 🙂 
동현님이 처음에 쓰셨던 것 처럼 flag를 써볼 수 있을 것 같아요. 단지 flag를 Controller의 인스턴스 변수로 두는 것 보다는 지역변수 정도로 둘 것 같네요. 🙂 사실 그 외에도 더 좋은 방법이 있을지는 모르겠지만 실패 케이스가 어느정도 누적되면 적당선에서 애플리케이션을 종료하도록하는 것도 하나의 방법일 것 같아요.